XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs

```diff
@@ -407,111 +407,18 @@ private void InitializeServiceLocator()
             {
-                if (TryCreateAndRegisterService<IMixedRealityCameraSystem>(ActiveProfile.CameraSystemType, out var service, ActiveProfile.CameraSystemProfile) && CameraSystem != null)
-                {
-                    TryRegisterDataProviderConfigurations(ActiveProfile.CameraSystemProfile.RegisteredServiceConfigurations, service);
-                }
-                else
+                if (configuration.Enabled)
                 {
-                    Debug.LogError("Failed to start the Camera System!");
-                }
-            }
-
-            if (ActiveProfile.IsInputSystemEnabled)
-            {
-                if (TryCreateAndRegisterService<IMixedRealityInputSystem>(ActiveProfile.InputSystemType, out var service, ActiveProfile.InputSystemProfile) && InputSystem != null)
-                {
-                    if (TryCreateAndRegisterService<IMixedRealityFocusProvider>(ActiveProfile.InputSystemProfile.FocusProviderType, out _))
-                    {
-                        TryRegisterDataProviderConfigurations(ActiveProfile.InputSystemProfile.RegisteredServiceConfigurations, service);
-                    }
-                    else
+                    if (TryCreateAndRegisterService(configuration, out var service) && service != null)
                     {
-                        inputSystem = null;
-                        Debug.LogError("Failed to register the focus provider! The input system will not function without it.");
+                        if (configuration.Profile is BaseMixedRealityServiceProfile<IMixedRealityDataProvider> profile)
+                        {
+                            TryRegisterDataProviderConfigurations(profile.RegisteredServiceConfigurations, service);
+                        }
                     }
                 }
-                else
-                {
-                    Debug.LogError("Failed to start the Input System!");
-                }
-            }
-
-            if (ActiveProfile.IsBoundarySystemEnabled)
-            {
-                if (TryCreateAndRegisterService<IMixedRealityBoundarySystem>(ActiveProfile.BoundarySystemSystemType, out var service, ActiveProfile.BoundarySystemProfile) && BoundarySystem != null)
-                {
-                    TryRegisterDataProviderConfigurations(ActiveProfile.BoundarySystemProfile.RegisteredServiceConfigurations, service);
-                }
-                else
-                {
-                    Debug.LogError("Failed to start the Boundary System!");
-                }
-            }
-
-            if (ActiveProfile.IsSpatialAwarenessSystemEnabled)
-            {
-#if UNITY_EDITOR
-                // Setup the default spatial awareness layers in the project settings.
-                LayerExtensions.SetupLayer(31, Definitions.SpatialAwarenessSystem.MixedRealitySpatialAwarenessSystemProfile.SpatialAwarenessMeshesLayerName);
```

Comment:
@FejZa we probably need to do something similar to how we setup the input axes but for the spatial awareness layers instead.